spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/AzureCloudConfigWatch.java

```diff
@@ -95,36 +98,43 @@ public void watchConfigKeyValues() {
         }
 
         for (ConfigStore configStore : configStores) {
-            if (needRefresh(configStore)) {
+            if (needRefreshConfiguration(configStore) || needRefreshFeatureFlag(configStore)) {
                 break;
             }
         }
     }
 
-    private boolean needRefresh(ConfigStore store) {
+    private boolean needRefreshConfiguration(ConfigStore store) {
         String watchedKeyNames = watchedKeyNames(store, storeContextsMap);
+        return needRefresh(store, CONFIGURATION_STORE, watchedKeyNames);
+    }
+    
+    private boolean needRefreshFeatureFlag(ConfigStore store) {
+        return needRefresh(store, FEATURE_STORE, FEATURE_STORE_WATCH_KEY);
+    }
+    
+    private boolean needRefresh(ConfigStore store, String storeName, String watchedKeyNames) {
         QueryOptions options = new QueryOptions().withKeyNames(watchedKeyNames)
                 .withLabels(store.getLabels()).withFields(QueryField.ETAG).withRange(0, 0);
 
         List<KeyValueItem> keyValueItems = configOperations.getRevisions(store.getName(), options);
+        
         if (keyValueItems.isEmpty()) {
             return false;
         }
 
         String etag = keyValueItems.get(0).getEtag();
-        if (firstTimeMap.get(store.getName()) == null) {
-            storeEtagMap.put(store.getName(), etag);
-            firstTimeMap.put(store.getName(), false);
+        if (firstTimeMap.get(store.getName() + storeName) == null) {
+            storeEtagMap.put(store.getName() + storeName, etag);
+            firstTimeMap.put(store.getName() + storeName, false);
             return false;
         }
 
-        if (!etag.equals(storeEtagMap.get(store.getName()))) {
-            LOGGER.trace("Some keys in store [{}] matching [{}] is updated, will send refresh event.",
-                    store.getName(), watchedKeyNames);
-            storeEtagMap.put(store.getName(), etag);
-            RefreshEventData eventData = new RefreshEventData(watchedKeyNames);
+        if (!etag.equals(storeEtagMap.get(store.getName() + storeName))) {
+            storeEtagMap.put(store.getName() + storeName, etag);
+            RefreshEventData eventData = new RefreshEventData(FEATURE_STORE_WATCH_KEY);
```

Comment:
The `new RefreshEventData(FEATURE_STORE_WATCH_KEY)` change is incorrect? Should still be `new RefreshEventData(watchedKeyNames)`.